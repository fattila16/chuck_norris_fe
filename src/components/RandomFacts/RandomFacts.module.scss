@import 'src/assets/scss/color-palette';
@import 'src/assets/scss/vars';

.container {
    margin-top: 40px;
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
    border-radius: 12px;
    box-shadow:0px 4px 18px rgba(0,0,0,0.1);
    background-color: map-get($active-theme , fawn );
    h2{
        padding: 40px;
        text-align: center;
        height: 120px;
    }
}
.button {
    margin-bottom: 20px;
}