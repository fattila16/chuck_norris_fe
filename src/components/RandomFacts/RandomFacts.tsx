import React from 'react';
import css from './RandomFacts.module.scss';
import useRandom from '../../hooks/useRandom';
import { useRouter } from 'next/dist/client/router';
import Button from '../Button/Button';

const RandomFacts: React.FC = () => {
    const router = useRouter();
    console.log(router);
    const { data: randomFact, error, refresh } = useRandom(router?.query?.category);
    if (error) return <>Error</>;
    return (
        <>
            <section className={css.container}>
                <h2>{randomFact?.value}</h2>
                <div className={css.button}>
                    <Button onClick={refresh} title="Another one" size="medium" />
                </div>
            </section>
        </>
    );
};

export default RandomFacts;
