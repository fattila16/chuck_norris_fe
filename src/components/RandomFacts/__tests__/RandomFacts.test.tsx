import React from 'react';
import { shallow } from 'enzyme';
import RandomFacts from '../RandomFacts';

describe('[RandomFacts]', () => {
    it('Should render RandomFacts component without error', () => {
        const component = shallow(<RandomFacts />);
        expect(component.getElements()).toMatchSnapshot();
    });
});
