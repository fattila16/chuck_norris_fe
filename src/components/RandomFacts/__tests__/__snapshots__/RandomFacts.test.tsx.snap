// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[RandomFacts] Should render RandomFacts component without error 1`] = `
Array [
  <React.Fragment>
    <section
      className="container"
    >
      <h2 />
      <div
        className="button"
      >
        <Button
          onClick={[Function]}
          size="medium"
          title="Another one"
        />
      </div>
    </section>
  </React.Fragment>,
]
`;
